pkg/tsdb/azuremonitor/azuremonitor-datasource.go

```diff
@@ -299,10 +297,46 @@ func (e *AzureMonitorDatasource) parseResponse(amr AzureMonitorResponse, query *
 			frame.SetRow(i, point.TimeStamp, value)
 		}
 
-		frames = append(frames, frame)
+		frameWithLink := addConfigData(*frame, getQueryUrl(query, azurePortalUrl))
+		frames = append(frames, &frameWithLink)
+	}
+
+	return frames
+}
+
+// Gets the deep link for the given query
+func getQueryUrl(query *AzureMonitorQuery, azurePortalUrl string) string {
+	aggregationType := aggregationTypeMap["Average"]
+	aggregation := query.Params.Get("aggregation")
+	if aggregation != "" {
+		if aggType, ok := aggregationTypeMap[aggregation]; ok {
+			aggregationType = aggType
+		}
 	}
 
-	return frames, nil
+	timeFrom := query.TimeRange.From.Format(time.RFC3339Nano)
+	timeTo := query.TimeRange.To.Format(time.RFC3339Nano)
+	timespan := fmt.Sprintf("{\"absolute\":{\"startTime\":%q,\"endTime\":%q}}", timeFrom, timeTo)
+	timespan = url.QueryEscape(timespan)
+
+	id := fmt.Sprintf("/subscriptions/%v/resourceGroups/%v/providers/%v/%v",
+		query.UrlComponents["subscription"],
+		query.UrlComponents["resourceGroup"],
+		query.UrlComponents["metricDefinition"],
+		query.UrlComponents["resourceName"],
+	)
+	chartDef := fmt.Sprintf(
+		"{\"v2charts\":[{\"metrics\":[{\"resourceMetadata\":{\"id\":%q},\"name\":%q,\"aggregationType\":%v,\"namespace\":%q,\"metricVisualization\":{\"displayName\":%q,\"resourceDisplayName\":%q}}]}]}",
```

Comment:
this is a bit error prone, I would define the struct so you can `json.Marshall` it. (the same for `timespan` while you are on it)